config/routes.rb

```diff
@@ -2,8 +2,10 @@
 
   root 'pages#show', page:'home'
 
-  resources :lessons
-  get '/lessons/publish/:id' => 'lessons#publish'
+  resources :lessons do
+    get 'publish', on: :member
```

Comment:
The `on: :member` is necessary to indicate that this route takes an ID, it doesn't work without it. Also I doubt checked and that path still passes the id as `params[:id]` so `find_lesson` still works. I checked by hand and it'll posts the correct lesson to twitter, so I don't think any changes are needed.